include/bbcode.php

```diff
@@ -1163,8 +1163,10 @@ function ($match) use ($simplehtml){
 	// fix any escaped ampersands that may have been converted into links
 	$Text = preg_replace('/\<([^>]*?)(src|href)=(.*?)\&amp\;(.*?)\>/ism', '<$1$2=$3&$4>', $Text);
 
-	// sanitizes src attributes (only relative redir URIs or http URLs)
-	$Text = preg_replace('#<([^>]*?)(src)="(?!http|redir|cid)(.*?)"(.*?)>#ism', '<$1$2=""$4 class="invalid-src" title="' . t('Invalid source protocol') . '">', $Text);
+	// sanitizes src attributes (http and redir URLs for displaying in a web page, cid used for inline images in emails)
+	$allowed_src_protocols = array('http', 'redir', 'cid');
```

Comment:
This won't change on each call of the method, maybe it is better to have it static?